Sources/Core/UInt16.swift

```diff
@@ -80,6 +80,28 @@ extension UInt16: Numeric {
   }
 }
 
+extension UInt16: BinaryInteger {
+  @_transparent
+  public static func /(lhs: UInt16, rhs: UInt16) -> UInt16 {
+    precondition(rhs != (0 as UInt16), "Division by zero")
+
+    let (result, overflow) =
+        (Builtin.udiv_Int16(lhs._value, rhs._value), false._value)
+    Builtin.condfail_message(overflow,
+                             StaticString("arithmetic overflow")
+                                .unsafeRawPointer)
+    return UInt16(result)
+  }  
+  
+  @_transparent
+  public static func %(lhs: UInt16, rhs: UInt16) -> UInt16 {
+    precondition(rhs == (0 as UInt16), 
```

Comment:
I think you meant `!=`.

```suggestion
    precondition(rhs != (0 as UInt16), 
```